### ISSUE TITLE
First invoke of module's (crypto) method is extremely slow

### ISSUE THREAD

**Issue by ghost** [electron/electron#2073]
I've noticed quite a big lag while adding first record in my application and after doing some tests it turned out that running `crypto.randomBytes()` for the first time takes incomparably longer than for all the consecutive runs. Here's the test code:

``` javascript
console.time('require');
var crypt = require('crypto');
console.timeEnd('require');

for (var i = 1; i < 4; i++) {
  console.time(i);
  crypt.randomBytes(10);
  console.timeEnd(i);
}
```

Running this with node produces:

```
require: 9ms
1: 0ms
2: 0ms
3: 0ms
```

and a simple electron app:

```
[988:0628/194243:INFO:CONSOLE(0)] require: 9ms
[988:0628/194243:INFO:CONSOLE(0)] 1: 525ms
[988:0628/194243:INFO:CONSOLE(0)] 2: 0ms
[988:0628/194243:INFO:CONSOLE(0)] 3: 0ms
```

and it's even longer if run within `browser-window`.

**Comment by zcbenz** [comment]
Which OS are you using?

**Comment by infensus** [comment]
Windows 8.1

**Comment by jrieken** [comment]
We have observed the same in VS Code (https://github.com/Microsoft/vscode/issues/22086) with times of around 1 second in the renderer. Also, we pay the price every time electron starts, whereas node isn't slow on subsequent starts.


Put the below snippet in `renderer.js` of the quick start sample, keep refreshing the renderer and see it always being slow, not (re-)using any sort of OS-level entropy

```js
let t1 = Date.now(); 
console.log(require('crypto').randomBytes(21).toString('hex')); 
console.log(`${Date.now() - t1}ms`);
```

**Comment by niik** [comment]
We've also observed the same behavior with initial load times of over a second on Windows 10. We spotted it using the `uuid` module (see https://github.com/kelektiv/node-uuid/issues/189#issuecomment-297663423)

We've [worked around it](https://gist.github.com/niik/d9a0991a8256cf92225c52e7a2800ad8) for now by using window.crypto as our source of entropy for `uuid` but since we could end up having one of our other dependencies source through the NodeJS crypto module it'd be real nice if we could solve this in electron proper.

**Comment by chgibb** [comment]
As a temporary work around, we've switched to using node-uuid's ```v4``` module as it does not invoke ```crypto.randomBytes``` at require time.

**Comment by Doccrazy** [comment]
I was able to work around the issue by initializing the crypto module on application start using the callback of the `crypto.randomBytes` method. This way, the initial load is performed in the background, without blocking the UI thread. A later synchronous call can then be made without delay. 
```js
setTimeout(() => {
  // asynchronously initialize the crypto library
  crypto.randomBytes(1, () => {
    // done
  });
}, 1000);
```
Still, a proper fix would be nice, esp. as the bug also applies to the RSA methods (`crypto.privateDecrypt` etc.).

**Comment by nornagon** [comment]
I can't reproduce this on Electron 3, 4 or 5. Here's the fiddle I'm using: https://gist.github.com/9b2e03eb35953a8e47342c7d9a9b94be

Is there something I'm missing? I tried this on Windows and Linux, and didn't observe any timings above ~20ms.

**Comment by zcbenz** [comment]
Probably silently fixed in upstream.

**Comment by nornagon** [comment]
Closing this for now, reopen if you find a way to reproduce it :)

**Comment by PriyankuBiswas-EH** [comment]
The issue is resolved in v32.1.2 as far i see. In version 26 it was very slow.
